terraform/modules/web/main.tf

```
@@ -0,0 +1,88 @@
+resource "aws_cloudfront_origin_access_control" "this" {
+  name                              = var.bucket.bucket_regional_domain_name
+  description                       = "Manages an AWS CloudFront Origin Access Control, which is used by CloudFront Distributions with an Amazon S3 bucket as the origin."
+  origin_access_control_origin_type = "s3"
+  signing_behavior                  = "always"
+  signing_protocol                  = "sigv4"
+}
+
+resource "aws_cloudfront_function" "viewer_request" {
+  count   = length(var.viewer_request_function_list)
+
+  code    = var.viewer_request_function_list[count.index].code
+  comment = var.viewer_request_function_list[count.index].comment
+  name    = var.viewer_request_function_list[count.index].name
+  runtime = var.viewer_request_function_list[count.index].runtime
+}
+
+resource "aws_cloudfront_distribution" "this" {
+  aliases             = [var.certificate.domain_name]
+  comment             = "Distribution for the ${var.certificate.domain_name} website"
+  default_root_object = "index.html"
+  enabled             = var.enabled
+  http_version        = "http2and3"
+  is_ipv6_enabled     = true
+  price_class         = "PriceClass_100"
+  web_acl_id          = var.web_acl.arn
+  
```

As I'm working through the security remediations, I notice one of the call-outs across the accounts is for CloudFront distributions to have logging enabled. Is this something we can enable by default here? Maybe have the bucket be passed in as part of the instantiation:
```suggestion
  logging_config {
    include_cookies = false
    bucket          = var.logging_bucket
  }
```

---

@jscott-nava I think we will still need to include a `logging_config` block in the distribution block for it to function